src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -1905,6 +1905,17 @@ public int compare(final ProTerritory t1, final ProTerritory t2) {
               continue;
             }
 
+            // Check if territory is safe after all current moves
+            final List<Unit> defendingUnits = moveMap.get(t).getAllDefenders();
```

Comment:
This needs a larger refactor as the parameters here and calc methods being called vary. But agree that it would be nice to simplify this 5-6 lines into a single method call that checks if battle result is null and if so then runs a calc method, sets result, and returns result.